### PR TITLE
refactor data extraction and loading logic and normalize database

### PR DESCRIPTION
**Pull request by kennethnym**
- data extraction and loading are now defined in a per-pmt module basis. for example, data extraction and loading logic for tenable is defined in a module called `tenable`
- the entry point for the data extraction and loading pipeline for each pmt is defined as simply a function called `pipeline` in a `pipeline.py` file located in the corresponding module (for tenable it's defined in `src/tenable/pipeline.py`. each pipeline function defines how data should be extracted from the corresponding pmt and how they should be processed and subsequently loaded into their corresponding database.
- the `main.py` now simply calls each aforementioned pipeline function sequentially (or in the future, concurrently via threading) for each pmt each pipeline function that needs to be called is neatly contained in a top level variable called `pipelines`.
- a proper database workflow has now also been set up, via [golang-migrate](https://github.com/golang-migrate/migrate)
  - because each pmt has its own database, their migrations are defined and handled separately. each pmt module contains a directory called `migrations` which contains all the migration SQL scripts for their own database. for tenable, the `migrations` directory is in `src/tenable/migrations`.
  - in the future, to evolve the database schema, simply create two SQL scripts, following the filename convention as outlined [here](https://github.com/golang-migrate/migrate/blob/master/MIGRATIONS.md) by golang-migrate
    - tldr: `{version}_{title}.up.{extension}` for defining schema modifications, and `{version}_{title}.down.{extension}` for reverting them. `version` is an increment of the current version. `title` is arbitrary but should be descriptive of what the migration is doing. `extension` should just be `sql`.
- the database has now also been normalized to 1nf. SQL queries defined in Grafana have been modified accordingly and tested in a VM I created.
  - on a sidenote, it might be worth investigating version controlling the Grafana dashboard JSON following [this guide on provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/), although if not many changes are anticipated in the future, this may not be necessary

### CODE REVIEW

**Comment by kennethnym** [comment]
this is still in draft because i still need to redo the now outdated database uml.

**Comment by kennethnym** [comment]
should be ready now. this is a big one, so feel free to hit me with any question you have :)

**Review by jcornall — CHANGES_REQUESTED** [review]
All looks good, spotted a couple typos in the .md files but apart from that these are great changes/additions

**Review by jcornall — APPROVED** [review]
All good to go
